device/globals/player.gd

```diff
@@ -317,6 +317,16 @@ func teleport_pos(x, y):
 	set_position(Vector2(x, y))
 	_update_terrain()
 
+func turn(deg):
+	if deg < 0 or deg > 360:
+		vm.report_errors("player", ["Invalid degree to turn " + String(deg)])
```

Comment:
Circles are divided into 360 degrees, there is no logic to turn the player round and round and round, and -1 is a special case in Escoria. Something like -2 doesn't make any sense, unless there's weird imaginary-number-type things going on.

Again, it's about robust code that tells the developer something's not right. So I say yes.

Reply:
Yes, a circle is divided into 360 degrees, but you can easily turn 1000 or -3000 degrees. This is not so useful now perhaps, but it could be if commands could take globals (which might be very hard to achieve) and a command let you turn an item/player by x degrees.

My question was more about how likely it is that a developer would use an invalid value though, to warrant this kind of check.

Reply:
I concede that you can rotate a dial by however many degrees and represent directions by using negative numbers. This does not, however, exist in Escoria. IMO if `turn_by` gets implemented, it should have this possibility and the animation support for it too.

In this case, where it does effectively "set" an angle (albeit with animations) there's no real reason to allow numbers. And -1 is still a special case.

The developer may put in a typo or have read the documentation a bit sloppily or want to see what happens. Especially junior hackers learn by trying to break the system.

What do you propose as an alternative?

Note also that doing an integer check is probably a CPU cycle's worth of computational power.

Reply:
I thought only a return value of -1 would be a special case; not input value of -1? I'm not awfully familiar with how this works, but my intuition was that this function would work just fine with any valid input and that if a value of < 0 or > 360 it would be for a reason, though it probably makes no difference at the moment.